ocaml/xapi/xapi_pool_transition.ml

```diff
@@ -24,12 +24,10 @@ open D
 
 (** Reset the role on disk, takes effect on next server restart only! *)
 let set_role r =
-  let open Pool_role in
-  let old_role = get_role () in
-  with_pool_role_lock (fun () ->
-      Unixext.write_string_to_file !Constants.pool_config_file (string_of r)) ;
-  Localdb.put Constants.this_node_just_became_master
-    (string_of_bool (old_role <> Master && r = Master))
+  let current_role = Pool_role.set_role_for_next_boot r in
+  (current_role <> Master && r = Master)
```

Comment:
Shouldn't the conditions be switched?

```suggestion
  (current_role = Master && r <> Master)
```
using `new_role` instead of `r` would make this clearer.